src/library_wasi.js

```diff
@@ -561,28 +561,24 @@ var WasiLibrary = {
   // random.h
 
   $initRandomFill: () => {
-    if (typeof crypto == 'object' && typeof crypto.getRandomValues == 'function') {
-#if SHARED_MEMORY
-      // like with most Web APIs, we can't use Web Crypto API directly on shared memory,
-      // so we need to create an intermediate buffer and copy it to the destination
-      return (view) => view.set(crypto.getRandomValues(new Uint8Array(view.byteLength)));
-#else
-      return (view) => crypto.getRandomValues(view);
-#endif
-    }
-
-#if ENVIRONMENT_MAY_BE_NODE
+#if ENVIRONMENT_MAY_BE_NODE && MIN_NODE_VERSION < 19
     if (ENVIRONMENT_IS_NODE) {
-      return (view) => require('crypto').randomFillSync(view);
+      // Node v19 and above has globalThis.crypto already available
+      crypto = require('crypto');
```

Comment:
Is this modifying the global scope intentionally?

Reply:
Yes, the code below that uses `crypto.getRandomValues` expects it to exist.

This is just the default behavior of modern node versions anyway.  I could add a check that it doesn't already exist perhaps?

Reply:
No, looks good, I just wanted to be sure I was reading it right.